### ISSUE TITLE
[💡 Feature]: add new mobile custom command to relaunch the app

### ISSUE THREAD

**Issue by wswebcreation** [webdriverio/webdriverio#12626]
### Is your feature request related to a problem?

The [`launchApp`](https://webdriver.io/docs/api/appium/#launchapp) command is deprecated and should be replaced with an alternative

### Describe the solution you'd like.

This is a sample we use for our native demo app. We need to check the bundleId/packageName of the current app under test.

```ts
async function relaunchApp() {
  const PACKAGE_NAME = 'com.wdiodemoapp'
  const BUNDLE_ID = 'org.reactjs.native.example.wdiodemoapp'
  const appIdentifier = browser.isAndroid ? { 'appId': PACKAGE_NAME } : { 'bundleId': BUNDLE_ID }
  const terminateCommand = 'mobile: terminateApp'
  const launchCommand = `mobile: ${driver.isAndroid ? 'activateApp' : 'launchApp'}`

  await browser.execute(terminateCommand, appIdentifier)
  await browser.execute(launchCommand, appIdentifier)
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jlipps** [comment]
I guess there's no reason it couldn't be in Appium itself. What do you think @mykola-mokhnach?

**Comment by mykola-mokhnach** [comment]
> I guess there's no reason it couldn't be in Appium itself. What do you think @mykola-mokhnach?

I don't see how such API could look like. In Android and iOS extensions to run apps work differently, not saying about other drivers.

For example, in order to restart an iOS app one could perform `mobile: launchApp` and xctest automatically terminates it if it is already running.
In Android the `mobile: startActivity` API one must set `stop` arg to `true` in order for the activity to be restarted upon launch.

**Comment by christian-bromann** [comment]
> I don't see how such API could look like.

Ideally it would be great if we could the operation of restarting an app within the refresh WebDriver command (`/session/{session id}/refresh`). To simplify this, it would require the `xcuitest` and `uiautomator2` driver to keep the information which app was or is currently opened. Can you confirm if these drivers are capable to do that? Maybe Appium could call `getCurrentActivity` for example in Android as part of the execution.

**Comment by mykola-mokhnach** [comment]
Unfortunately it's not so simple. The current activity might not be launchable. Also, we don't know what/if we need to apply any additional arguments or environment variables on startup.

**Comment by wswebcreation** [comment]
It took some time, sorry 😅 


@christian-bromann , what do you expect from a "refresh"?


I'm now looking into creating a custom command for WebdriverIO for this so it can be added to the mobile commands. If I understand the docs correctly, I can do this for iOS

```ts
// See https://appium.github.io/appium-xcuitest-driver/latest/reference/execute-methods/#mobile-launchapp
const { bundleId, processArguments: { args, env } } = await browser.execute('mobile: activeAppInfo') as ActiveAppInfo
const iOSLaunchOptions = {
     bundleId,
     ...(args.length > 0 && { arguments: args }),
     ...(Object.keys(env).length > 0 && { environment: env })
}
await browser.execute('mobile: terminateApp', { bundleId })

await browser.execute('mobile:launchApp', iOSLaunchOptions)
```

This would 

- retrieve all the info of the "active running app", including all provided args/environment data to start it, 
- terminate the app, 
- start it again with the initial loaded args/environment values. 

I can't find a way to "reset" the app state.

So the above would only restart the app the way it was initially started. Resetting the state would be the responsibility of the user.

For Android we can do this

```ts
const packageName = await browser.getCurrentPackage()
const activity = await browser.getCurrentActivity()

// https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-terminateapp
await browser.execute('mobile: terminateApp', { appId: packageName })
// Delete all data associated with a package
// https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-clearapp
await browser.execute('mobile: clearApp', { appId: packageName })

// Or use this
// This starts the app as if you would click the icon, nothing more
await browser.execute('mobile: activateApp', { appId: packageName })

// Or use something like this
await browser.execute('mobile: startActivity', { intent: `${packageName}/${activity} })
```

This would do the following for Android:
- get the packageName of the current running app, including the activity
- Termine the app
- clear all the app data
- OR:
  - start the app as if you would click the app icon and, assuming, start the first screen it should load
  - start the app with the latest running activity, this might break if this activity needs data that has been removed due to the clear


@mykola-mokhnach , am I correct based on the docs?


My suggestion would be to create a `restartApp` method that would close the app and start it again with the arguments it was started before. This would mean:
- iOS: also provide the env/args
- Android: don't clear the data, and start the app as if you would click on the icon. this would mean that you would miss `optionalIntentArguments` or `intentCategory` which can be retrieved from the `browser.requestedCapabilities`

**Comment by mykola-mokhnach** [comment]
> @mykola-mokhnach, am I correct based on the docs?

That looks correct to me. Unfortunately, on iOS there is no possibility to reset application data except of completely reinstalling it.

As for Android case the app might be executed with additional intent arguments or a custom intent name. Not sure how these scenarios could be covered though.

**Comment by wswebcreation** [comment]
Yeah, was also looking into that for Android. I might be able to pull that out of the requested caps, but if it has been started in the middle of the session we can't get it unless we store it somewhere 

I think I would go for a "simple" approach 

Tnx @mykola-mokhnach !
